include/librealsense2/h/rs_option.h

```diff
@@ -340,6 +340,22 @@ extern "C" {
     */
     const char* rs2_get_option_value_description(const rs2_options* options, rs2_option option, float value, rs2_error ** error);
 
+    /**
+    * sets a callback in case an option in this options container value is updated
```

Comment:
Captalization (`Set`).
I think you want to mention the cost: that this creates a thread per container that periodically queries the options.
We also have to be careful: it **sounds** like this will apply to all options, where in reality (at least now) it applies to only specific options.

Reply:
Fixed capitalization and added extra thread warning.
Did not address all/some options. Do you have a recommended explanation? I did not want to add anything we will need to remove later and might forget.
On the same lines, I considered adding a way for the user to register wanted options to watch, but did not do it yet as it might not be needed later and we might not want to remove it from the API.